src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,204 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.File;
+
+import duke.exception.IllegalCommandException;
+import duke.exception.EmptyCommandException;
+import duke.task.Task;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+public class Duke {
+
+    public static ArrayList<Task> tasks = new ArrayList<>();
+    public static final String FILE_NAME = "duke.txt";
+    public static final String DIR_NAME = "data";
+    public static final String FILE_PATH = DIR_NAME + "/" + FILE_NAME;
+
+    /**
+     * Prints a farewell message when the user exits the program.
+     */
+    public static void commandBye() {
+        System.out.println("Bye! I hope to see you again soon :)");
+    }
+
+    /**
+     * Lists all tasks with their status.
+     */
+    public static void commandList() {
+        System.out.println("Here are your current tasks and their status:");
+        for (Task task : tasks) {
+            System.out.println((tasks.indexOf(task) + 1) + ". " + task);
+        }
+    }
+
+    /**
+     * Adds a task of a certain type defined by the user to the list.
+     *
+     * @param line The command entered by the user.
+     */
+    public static void addTask(String line) throws EmptyCommandException, IllegalCommandException, IOException {
+
+        if (!line.startsWith("todo") && !line.startsWith("deadline") && !line.startsWith("event")) {
```

Comment:
Quite a [complex and long condition](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions) here. Maybe better to extract them out as variables first before checking?